pennylane/devices/default_qubit.py

```diff
@@ -132,6 +164,11 @@ def _apply_operation(self, operation):
             self._apply_basis_state(operation.parameters[0], wires)
             return
 
+        apply_func = self._ops_map.get(operation.name, None)
+        if apply_func:
+            self._state = apply_func(self._state, axes, inverse=operation.inverse)
+            return
```

Comment:
Since these are methods, I would suggest removing the need to pass both `self._state` and return the state:

```suggestion
        apply_func = self._ops_map.get(operation.name, None)
        if apply_func:
            apply_func(axes, inverse=operation.inverse)
            return
```

This is also more consistent with the rest of default.qubit.

Slight nitpick, but I tend to prefer the following pattern for dispatching functions from dictionaries, in being easier to read/parse:

```python
if operation.name in self._ops_map:
    self._ops_map[operation.name](axes, inverse=operation.inverse)
```

There is additional overhead in that the dictionary is being accessed twice (which is fixed with the walrus operator in Python 3.8 😆), but I think this might be negligible?